### PR TITLE
fix: update stage backend permission by cr

### PR DESCRIPTION
**Pull request by zhu327**
### Description

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/dashboard/apigateway/apigateway/apis/open/permission/serializers.py [hunk 1]
```diff
@@ -37,15 +37,15 @@
...
     target_app_code = serializers.CharField(max_length=32, required=True)
 
 
-class AppAPIPermissionQuerySLZ(serializers.Serializer):
+class AppGatewayPermissionInputSLZ(serializers.Serializer):
     target_app_code = serializers.CharField(max_length=32, required=True)
 
 
-class AppPermissionResourceSLZ(serializers.Serializer):
+class AppPermissionResourceOutputSLZ(serializers.Serializer):
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/apis/open/permission/serializers.py:48]: AppResourcePermissionOutputSLZ

File: src/dashboard/apigateway/apigateway/apis/open/permission/views.py [hunk 2]
```diff
@@ -53,19 +57,29 @@
 logger = logging.getLogger(__name__)
 
 
+def get_permission_model(dimension: str):
+    if dimension == GrantDimensionEnum.API.value:
+        return AppAPIPermission
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/apis/open/permission/views.py:62]: to AppGatewayPermission ?

File: src/dashboard/apigateway/apigateway/apis/web/permission/serializers.py [hunk 3]
```diff
@@ -46,16 +44,32 @@ class AppGatewayPermissionOutputSLZ(serializers.Serializer):
     renewable = serializers.SerializerMethodField()
 
     def get_resource_id(self, obj):
-        return obj.resource.id if hasattr(obj, "resource") else 0
+        if not hasattr(obj, "resource_id"):
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/apis/web/permission/serializers.py:47]: 如下修改，会更明确吗？

AppGatewayPermissionOutputSLZ 中
```
def get_resource_id(self, obj):
    return 0
```

AppResourcePermissionOutputSLZ 中，继承此方法
```
def get_resource_id(self, obj):
    resource = self.context.get("resource_map", {}).get(obj.resource_id)
    return resource.id if resource else 0
```

**Review by alex-smile — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/apis/web/permission/serializers.py [hunk 4]
```diff
@@ -46,16 +44,32 @@ class AppGatewayPermissionOutputSLZ(serializers.Serializer):
     renewable = serializers.SerializerMethodField()
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/apis/web/permission/serializers.py:44]: NIT: maybe, 字段改成 resource = serializers.SerializerMethodField()，下面的方法好些一点点

**Review by Han-Ya-Jun — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/apis/open/permission/serializers.py [hunk 5]
```diff
@@ -37,15 +37,15 @@
 from apigateway.utils import time
 
 
-class AppPermissionResourceQuerySLZ(serializers.Serializer):
+class AppPermissionResourceInputSLZ(serializers.Serializer):
```
[the remark this answers is not in the document]
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/apis/open/permission/serializers.py]: Done

File: src/dashboard/apigateway/apigateway/apis/open/permission/serializers.py (hunk 1 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/apis/open/permission/serializers.py]: Done

File: src/dashboard/apigateway/apigateway/apis/open/permission/views.py (hunk 2 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/apis/open/permission/views.py]: Done

File: src/dashboard/apigateway/apigateway/apis/web/permission/serializers.py (hunk 3 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/apis/web/permission/serializers.py]: Done

File: src/dashboard/apigateway/apigateway/apis/open/stage/serializers.py [hunk 6]
```diff
@@ -229,6 +278,15 @@ def create(self, validated_data):
             instance.gateway_id, PluginBindingScopeEnum.STAGE.value, instance.id, stage_config
         )
 
+        # 3. 需要发布
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/apis/open/stage/serializers.py:281]: 去掉发布

**Review by wklken — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/apis/open/stage/serializers.py (hunk 6 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/apis/open/stage/serializers.py]: Done

**Pull request merged**